### PR TITLE
Improve API support for disposition objects.

### PR DESCRIPTION
**Pull request by phgross**
This PR adds some API improvements for dispositions, to be able to have translated and speaking validation messages in the UI.

Definition of Done: https://4teamwork.atlassian.net/wiki/spaces/CHX4TW/pages/917562/

For [CA-2116]

## Checklist
- [x] Changelog entry
- [x] Link to issue (Jira or GitHub) and backlink in issue (Jira)
- Upgrade steps (changes in profile):
  - [ ] Make it deferrable if possible _(not necessary only profile update)_
- New translations
  - [x] All msg-strings are unicode

[CA-2116]: https://4teamwork.atlassian.net/browse/CA-2116?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: opengever/disposition/validators.py [hunk 1]
```diff
@@ -38,6 +36,7 @@ def validate(self, value):
                       'different disposition.',
                       mapping={'title': dossier.title}))
 
+        return super(OfferedDossiersValidator, self).validate(value)
```
Comment (njohner) [inline opengever/disposition/validators.py:45]: seems to be missing a new line after the end of the method definition here.

File: opengever/disposition/validators.py [hunk 2]
```diff
@@ -37,6 +43,10 @@ def validate(self, value):
                       mapping={'title': dossier.title}))
 
         return super(OfferedDossiersValidator, self).validate(value)
+
+    def is_active(self, dossier):
+        return api.content.get_state(dossier) == 'dossier-state-active'
```
Comment (njohner) [inline opengever/disposition/validators.py:48]: `return dossier.is_open()` ?

File: opengever/disposition/locales/fr/LC_MESSAGES/opengever.disposition.po [hunk 3]
```diff
@@ -43,6 +43,11 @@ msgstr "Numéro courant"
 msgid "edit_transfer_number"
 msgstr "Editer le numéro de livraison"
 
+#. Default: "The dossier ${title} is still active."
+#: ./opengever/disposition/validators.py
+msgid "error_dossier_is_active"
+msgstr ""
```
Comment (njohner) [inline opengever/disposition/locales/fr/LC_MESSAGES/opengever.disposition.po:49]: "Le dossier ${title} est encore actif."

File: opengever/disposition/tests/test_workflow.py [hunk 4]
```diff
@@ -169,6 +169,25 @@ def test_appraising_is_not_possible_if_the_appraisal_is_incomplete(self, browser
...
+    def test_appraising_via_api_is_not_possible_if_the_appraisal_is_incomplete(self, browser):
+        self.login(self.archivist, browser)
+        appraisal = IAppraisal(self.disposition)
+        appraisal.update(dossier=self.offered_dossier_to_archive, archive=None)
+
+        url = '{}/@workflow/disposition-transition-appraise'.format(
+            self.disposition.absolute_url())
+
+        with browser.expect_http_error(code=500):
+            browser.open(url, method='POST', headers=self.api_headers)
```
Comment (njohner) [inline opengever/disposition/tests/test_workflow.py:182]: Shouldn't that go into the API tests? And is 500 the correct code to return here? Is that what other workflow transitions also return?

**Review by njohner — APPROVED** [review]
Apart from a few comments, LGTM

File: opengever/disposition/locales/fr/LC_MESSAGES/opengever.disposition.po (hunk 3 above)
Comment (phgross) [reply opengever/disposition/locales/fr/LC_MESSAGES/opengever.disposition.po:49]: thanks

File: opengever/disposition/validators.py (hunk 1 above)
Comment (phgross) [reply opengever/disposition/validators.py:45]: double checked, there is a new line.

File: opengever/disposition/tests/test_workflow.py (hunk 4 above)
Comment (phgross) [reply opengever/disposition/tests/test_workflow.py:182]: 👍 you're right, changed to 403 Forbidden.

**Pull request merged**